planner/core/cache.go

```diff
@@ -130,15 +130,21 @@ func SetPstmtIDSchemaVersion(key kvcache.Key, pstmtID uint32, schemaVersion int6
 }
 
 // NewPlanCacheKey creates a new planCacheKey object.
-func NewPlanCacheKey(sessionVars *variable.SessionVars, pstmtID uint32, schemaVersion int64) kvcache.Key {
+func NewPlanCacheKey(sessionVars *variable.SessionVars, stmtText, stmtDB string, schemaVersion int64) (kvcache.Key, error) {
+	if stmtText == "" {
+		return nil, errors.New("no statement text")
+	}
+	if stmtDB == "" {
+		return nil, errors.New("no statement database")
+	}
```

Comment:
Yes, this could happen. But I think this case seems trivial, and returning an error in this case for safety is acceptable. What do you think? @Reminiscent 
```
mysql> prepare stmt from 'select * from test.t';
Query OK, 0 rows affected (0.00 sec)

mysql> execute stmt;
ERROR 1105 (HY000): no statement database
```

Reply:
I think it seems better to use the normal optimize process rather than return an error.